### DISCUSSION TITLE
Flutter riverpod: initial value of StateProvider variable set to null

### DISCUSSION THREAD

**Discussion by BenzDroid** [rrousselGit/riverpod#2483] [Q&A] [answered]
Upon now I always use riverpod observable declarations like this:

final versionProvider = StateProvider<String>((ref) => "");
final totalPercentageProvider = StateProvider<int>((ref) => 0);

But what if some variables don't have an initial value at first, and I actually want them to be null and later change them?

**Comment by what2003** [comment]
then why don't 
final versionProvider = StateProvider((ref) =>null);

**Comment by AhmedLSayed9** [comment, marked as answer]
```dart
final versionProvider = StateProvider<String?>((ref) => null);
final totalPercentageProvider = StateProvider<int?>((ref) => null);
```

**Reply by BenzDroid** [reply]
error: The return type 'Null' isn't a 'String', as required by the closure's context.

**Reply by BenzDroid** [reply]
final versionProvider = StateProvider<String?>((ref) => null); work now
thank you
